doc/releases/migration-guide-3.6.rst

```diff
@@ -14,6 +14,23 @@ the :ref:`release notes<zephyr_3.6>`.
 Required changes
 ****************
 
+Architecture
+============
```

Comment:
This needs to go into the migration guide for v3.7.0 instead.

Reply:
rebased